### PR TITLE
Fixed removed button, fixed adding button, removed paging system, and fixed empty row bug 

### PR DESCRIPTION
**Pull request by knick1132**
ty D

### CODE REVIEW

File: grid.html [hunk 1]
```diff
@@ -35,7 +35,7 @@ <h2>Save plan as: </h2>
       <button onclick="removeRow()" class="red ghost">Remove Row</button>
       <button id="toggleSuperuserControls" class="pink ghost">Edit Header</button>
       <p></p>
-      <div id="propContainer">
+      <!--<div id="propContainer">
```
Comment (darylpuppy) [inline grid.html]: Delete things that are commented out

File: resources/js/file.js [hunk 2]
```diff
@@ -70,9 +70,11 @@ function savePlan() {
 
   // Converts the grid's data to a CSV, then to a JSON for export
   var rowData = exportCSV();
-  rowData = convertCSV(rowData);
+  rowData = JSON.parse(convertCSV(rowData));
+  rowData.pop();
+  rowData = JSON.stringify(rowData);
   //console.log("Rowdata:");
```
Comment (darylpuppy) [inline resources/js/file.js]: Delete console.logs

File: resources/js/main.js [hunk 3]
```diff
@@ -114,7 +114,7 @@ function numOfRows() {
 function addRow() {
   // Calculate the number of "pages" for the grid
   // For application, these probably represent months
-  numOfPages = (numOfRows()) / pageSize;
+  //numOfPages = (numOfRows()) / pageSize;
```
Comment (darylpuppy) [inline resources/js/main.js]: delete

File: resources/js/main.js [hunk 4]
```diff
@@ -123,14 +123,10 @@ function addRow() {
   };
 
   // Add a new row to each page at incrementing index (see below)
-  for(i = 1; i <= numOfPages; i++) {
-    index = (pageSize * i) + t; // to insert at same relative pos. for each page
-    addRowAtIndex(index, newRow);
-    t += 1; // increment at each page to account for the new row
-  }
+  index = i ; // to insert at same relative pos. for each page
```
Comment (darylpuppy) [inline resources/js/main.js:119]: What is i?

File: resources/js/main.js [hunk 5]
```diff
@@ -152,6 +148,7 @@ function removeRowAtIndex(targetIndex) {
     if(rowNode.index == targetIndex) {
         console.log("Removing row "+index);
         gridOptions.api.updateRowData({remove: [rowNode.data]});
+
```
Comment (darylpuppy) [inline resources/js/main.js]: remove this row

File: resources/js/main.js [hunk 6]
```diff
@@ -180,24 +177,20 @@ function removeRow(selectedRow) {
...
-  for(i = 1; i <= numOfPages; i++) {
-    console.log("deleting rows " + index)
-    console.log("deleting rowID " + selectedRow)
-    //console.log("row Data: " + selectedRow.data)
-    removeRowAtIndex(index);
-    index = (index + pageSize + i);
-    
-  }
-
+  //console.log("row Data: " + selectedRow.data)
```
Comment (darylpuppy) [inline resources/js/main.js]: get rid of this line.

File: resources/js/main.js [hunk 7]
```diff
@@ -180,24 +177,20 @@ function removeRow(selectedRow) {
   // Find the "relative" row location for each page and remove rows
   // from page 1 onward
   //index = index % pageSize;
-  
+  var selectedData = gridOptions.api.getSelectedRows();
+  var res = gridOptions.api.updateRowData({remove: selectedData});
 
   // Calculate number of *full* pages in the plan
   numOfPages = Math.floor((numOfRows()) / pageSize);
```
Comment (darylpuppy) [inline resources/js/main.js]: Why is numOfPages still being modified if there are no pages?

File: resources/js/main.js [hunk 8]
```diff
@@ -180,24 +177,20 @@ function removeRow(selectedRow) {
...
-    index = (index + pageSize + i);
-    
-  }
-
+  //console.log("row Data: " + selectedRow.data)
+  removeRowAtIndex(index);
   // Decrement the number of rows to show per page and update grid
-  pageSize -= 1;
-  gridOptions.api.paginationSetPageSize(Number(pageSize));
+
```
Comment (darylpuppy) [inline resources/js/main.js]: Why are there so many empty lines?

File: resources/js/main.js [hunk 9]
```diff
@@ -180,24 +177,20 @@ function removeRow(selectedRow) {
...
-    console.log("deleting rows " + index)
-    console.log("deleting rowID " + selectedRow)
-    //console.log("row Data: " + selectedRow.data)
-    removeRowAtIndex(index);
-    index = (index + pageSize + i);
-    
-  }
-
+  //console.log("row Data: " + selectedRow.data)
+  removeRowAtIndex(index);
```
Comment (darylpuppy) [inline resources/js/main.js]: Why are you both updating the ag-grid to remove rows and calling this method?

**Comment by darylpuppy** [comment]
There should be a confirmation for when changes are saved.

File: resources/js/main.js (hunk 4 above)
Comment (knick1132) [reply resources/js/main.js:119]: It just works for some reason..im not sure why

**Pull request closed without merging**